apps/oc_chef_wm/src/chef_wm_search.erl

```diff
@@ -156,6 +156,39 @@ solr_ids_length({Solr1Ids, _}) ->
 solr_ids_length(Solr1Ids) ->
     length(Solr1Ids).
 
+%% @doc Check the READ authz permissions on a list of search results, returning only the
+%% list of available results
+filter_permitted_results(_ReqId, _RequestorId, _OrgId, _DbContext, _IndexType, []) ->
+    [];
+filter_permitted_results(ReqId, RequestorId, OrgId, DbContext, {data_bag, BagName}, Ids) ->
+    case chef_db:fetch(#chef_data_bag{org_id = OrgId, name = BagName}, DbContext) of
+        not_found -> [];
+        #chef_data_bag{authz_id = AuthzId} ->
+            case oc_chef_authz:is_authorized_on_resource(RequestorId, object, AuthzId, actor, RequestorId, read) of
+                false -> [];
+                true -> Ids;
+                Error ->
+                    lager:error("is_authorized_on_resource failed (~p, ~p, ~p): ~p~n",
+                                [read, {data_bag, BagName}, RequestorId, Error]),
+                    {{halt, 500}, ReqId}
+            end
+    end;
+filter_permitted_results(ReqId, RequestorId, _OrgId, DbContext, IndexType, Ids) ->
+    Auth = chef_db:bulk_get_authz_ids(DbContext, IndexType, Ids),
+    AuthzIds = [ {AuthzId, Id} || [Id, AuthzId] <- Auth ],
+    Type = case IndexType of
+               client -> actor;
+               _ -> object
+           end,
+    case ?SH_TIME(ReqId, oc_chef_authz, bulk_actor_is_authorized, (ReqId, RequestorId, Type, AuthzIds, read)) of
+        true ->
+            Ids;
+        {false, NoAuthzList} ->
+            [ Id || Id <- Ids, not lists:member(Id, NoAuthzList)];
```

Comment:
@marcparadise my erlang is very raw, so am totally open to suggestions on how to do better than `lists:member`